text/3759-cargo-supported-targets.md

```diff
@@ -0,0 +1,592 @@
+Also, this is new syntax not currently used by cargo.
+
+### Allowing only target-tuples
+
+This is an even stricter version of the above. Set relations between `supported-targets` lists are
+exact, and the resolver can determine if a platform-specific dependency can be pruned from the
+dependency tree more easily, hence why the original proposal chose this format. Being even simpler
+to implement, this alternative may not be expressive enough for the common use case. Packages rarely
+support specific target-tuples, rather they support/require specific target attributes. What would
+likely happen is that packages would copy and paste the target-tuple list matching their
+requirements from somewhere or someone else. Every time a new target with the same attribute is
+added, the whole ecosystem would have to be updated.
+
+# Prior art
+[prior-art]: #prior-art
+
+Users can already select which packages they want to select in a workspace with the flags
+`--package` and `--exclude`. Cargo features can also be used to restrict which cargo-target
+is built using the `required-features` field. However, `required-features` does not allow filtering
+packages in a workspace, nor does it allow filtering out the library of a package.
+
+The `per-package-target` nightly feature defines the `force-target` field, which is supposed to
+force the package to build for a specific target-tuple. This does not interact well when used in
+dependencies, as one would expect a dependency to be built for the same target as the package.
+`supported-targets` supersedes `force-target` because instead of enforcing a single target, it
+enforces a set of targets.
+
+Published crates have mainly used their documentation to specify which targets they support, or they
+would leave it up to the user to infer it. Some crates also made use of compile time errors to
+ensure that `cfg` requirements are met, for example:
+```rust
+#[cfg(not(any(…)))]
+compile_error!("unsupported target cfg");
+```
+[`getrandom`](https://github.com/rust-random/getrandom/blob/9fb4a9a2481018e4ab58d597ecd167a609033149/src/backends.rs#L156-L160)
+is an example of a crate utilizing this method.
+
+In other system level languages, vendoring dependencies is a common practice, and the user would be
+responsible for ensuring that the dependencies are compatible with the target.
+
+Some higher-level languages and build tools have the ability to specify which platforms are compatible.
+- Python has [platform compatibility tags](https://packaging.python.org/en/latest/specifications/platform-compatibility-tags/#platform-compatibility-tags).
+    The reference explains how these are [used](https://packaging.python.org/en/latest/specifications/platform-compatibility-tags/#use)
+    by installers to determine which build of a package to install. PyPI also has [classifiers](https://pypi.org/classifiers/)
+    which allows to add metadata (including which platforms are allowed) to a package.
```

Comment:
Generally with bullets like this, people don't expect new information to be added and can miss classifiers being mentioned in here.  Better to have it as its own item
```suggestion
- Python package has [classifiers](https://pypi.org/classifiers/) as package metadata that includes supported platforms
- Python wheels (pre-built packages) have [platform compatibility tags](https://packaging.python.org/en/latest/specifications/platform-compatibility-tags/#platform-compatibility-tags).
    The reference explains how these are [used](https://packaging.python.org/en/latest/specifications/platform-compatibility-tags/#use)
    by installers to determine which build of a package to install.
```